Automattic/wp-calypso
# Enable Twitter Ads tracker.

#### Changes proposed in this Pull Request

* Enable Twitter Ads tracker (pau2Xa-2gU-p2)

#### Testing instructions

- Ensure the appropriate flags are set to enable analytics and tracking
  - You can use something like the Requestly extension to set cookies that enable flags based on the Request URL
  - Request URL Regex: `/^https?\:\/\/(?:[^\/]+\.)?(?:wordpress\.com|calypso\.live|calypso\.localhost\:3000)\/(?:start|plans|checkout|checklist|jetpack\/connect)(?:$|[\/?&#])/i`
  - Cookies to set (also enables Calypso debug messages in console; enabling `sensitive_pixel_option` is the same as accepting the cookie banner in GDPR countries):
```
document.cookie = 'flags=gdpr-banner,google-analytics,ad-tracking,a8c-analytics.on; path=/; domain=.' + document.location.hostname.split( '.' ).slice( -2 ).join( '.' );
document.cookie = 'sensitive_pixel_option=yes; path=/; domain=.' + document.location.hostname.split( '.' ).slice( -2 ).join( '.' );
localStorage.setItem( 'debug', 'calypso:*' );
```

- Load calypso.live test URL
- Enable Preserve Log on Chrome Network tab
- Visit `/start/user` and create a new account
- Verify there are no JS errors in the browser console (`idpiframe` and `Happychat`-related errors are expected due to testing environment)
- Verify `static.ads-twitter.com/uwt.js` is present in Network tab
- Verify `nvzbs` is present in Network tab (Twitter Ads)
- Verify there's a request to `https://analytics.twitter.com/i/adsct` with an `events: [["signup",{}]]` query parameter
- Verify there's a request to `https://analytics.twitter.com/i/adsct` with an `events: [["pageview",{}]]` query parameter
- Verify there's a request to `https://t.co/i/adsct` with an `events: [["signup",{}]]` query parameter
- Verify there's a request to `https://t.co/i/adsct` with an `events: [["pageview",{}]]` query parameter

- For good measure, confirm that the `calypso_page_view` event is firing in Network tab
